src/main/java/com/github/onsdigital/babbage/api/endpoint/content/Resource.java

```diff
@@ -23,13 +31,63 @@
     @GET
     public void get(@Context HttpServletRequest request, @Context HttpServletResponse response) throws IOException {
         try {
-            ContentResponse contentResponse = ContentClient.getInstance().getResource(request.getParameter("uri"));
+            String uri = request.getParameter("uri");
+            String width = request.getParameter("width");
+
+            ContentResponse contentResponse = ContentClient.getInstance().getResource(uri);
             String contentDispositionHeader = "inline; ";
             contentDispositionHeader += contentResponse.getName() == null ? "" : "filename=\"" + contentResponse.getName() + "\"";
             response.setHeader("Content-Disposition", contentDispositionHeader);
-            new BabbageContentBasedBinaryResponse(contentResponse, contentResponse.getDataStream(), contentResponse.getMimeType()).apply(request, response);
+
+            try ( InputStream contentResponseBody = contentResponse.getDataStream() ) {
+
+                BufferedImage image = null;
+
+                if (width != null) {
+
+                    try {
+                        Integer w = Integer.parseInt(width, 10);
+
+                        if (uri.toLowerCase().endsWith(".png") || uri.toLowerCase().endsWith(".jpg") || uri.toLowerCase().endsWith(".jpeg")) {
```

Comment:
I still think we should check `uri` is not null as this will cause a null pointer exception on this line if it is.

Reply:
As discussed - if these are empty or null then then appropriate error response is returned from the content client and it will never get this far.